app/controllers/staff/activities_controller.rb

```diff
@@ -39,7 +41,25 @@ def id
   end
 
   def organisation_id
-    params[:organisation_id]
+    organisation_id = params.fetch(:organisation_id, current_user.organisation)
+
+    return current_user.organisation.id unless known_organisation_id?(organisation_id)
+    return current_user.organisation.id if organisation_id.blank?
+    return current_user.organisation.id if requested_organisation_is_not_current_users?(organisation_id) && current_user_is_not_service_owner?
+    organisation_id
+  end
+
+  def requested_organisation_is_not_current_users?(requested_organisation_id)
+    requested_organisation_id != current_user.organisation.id
+  end
+
+  def current_user_is_not_service_owner?
+    return false if current_user.organisation.service_owner?
```

Comment:
You could replace this with `!current_user.organisation.service_owner?`

Reply:
😄 It funny raising this as I debated this with myself a great deal! 

I find the bang infront of something a little hard to approach, it's so easily missed. Ideally I would have a `delivery_partner?` method as I think that allows the check to be for a postitive, which feels nice and seems to read clearer in Ruby, but that seemed like a push too far.

Happy to change this to the bang if you would like it, would you keep the method or just drop it inline?

Reply:
I think there are plenty of places where `delivery_partner?` would be useful. Perhaps a method belongs inside `app/models/organisation.rb`? Then you could use it inline rather than having a separate method in here.

Reply:
@leeky I've gone for the User model as this is where the existing `service_owner?` method lives - let me know what you think?

Reply:
@mec Perfect 🎉